src/networking.c

```diff
@@ -3246,8 +3247,10 @@ char *getClientSockname(client *c) {
 int isClientConnIpV6(client *c) {
     /* The cached client peer id is on the form "[IPv6]:port" for IPv6
      * addresses, so we just check for '[' here. */
-    if ((c->conn == NULL || c->conn->type == NULL) && server.current_client) {
-        /* Fake client? Use current client instead. */
+    if (c->flag.fake && server.current_client) {
+        /* Fake client? Use current client instead.
```

Comment:
as far as this function is concerned, I don't think it should fall back to `current_client`. I think we should either let the caller to make the call on whether to fall back to the `current_client` or change the function name to better capture the fallback behavior.

@zuiderkwast thoughts?

Reply:
Obviously RediSearch is calling CLUSTER SLOTS from a module. Which IPs do we want here?

I think we should behave the same as we do for TLS ports. Do we return TLS ports in CLUSTER SLOTS if the `current_client` is using TLS?

Reply:
From a quick look at the source code, it looks like were returning TLS port according to the `current_client`'s TLS connection, using `shouldReturnTlsInfo()`, in these cases:

* CLUSTER NODES
* MOVED and ASK redirects
* CLUSTER REPLICAS

But I can't see that we do this for CLUSTER SLOTS. Here it looks like we're returning the non-TLS port to fake clients, if we have a cached such response. We only check `connIsTLS(c->conn)` here and no conn means no TLS. This looks like a bug though, because if the response is not cached, we use `shouldReturnTlsInfo()` when we generate the response and then we might cache it for the wrong parameters. The fake client is initiated with `client *createCachedResponseClient(int resp)` which only takes the RESP version, but TLS and IP-family are taken from global state... beautiful mess.

In CLUSTER SHARDS we return both ports in two separate fields. (We should probably add both IPv4 and IPv6 here, and the internal cluster bus IP too, on separate fields.)

Conclusion: I think it's correct to return IPv6 if the `current_client` uses it.

Reply:
I have confirmed this bug, but it can only be fixed when PR is merged, because otherwise we will crash on the IPv6 check.

If we do CLUSTER SLOTS from EVAL over TLS, it fails at `debugServerAssertWithInfo(c, NULL, verifyCachedClusterSlotsResponse(cached_reply, c->resp) == 1);` in `clusterCommandSlots`.

Reply:
> beautiful mess.

Can't agree more ...

Thanks @zuiderkwast! I am convinced by you that we should fall back to `current_user` and we can keep the existing name. Also great catch on `CLUSTER SLOTS`.

Reply:
i thought #581 already handle the TLS issue.

ohh, ok i see the problem.. It is indeed a mess. So lets fix it after this PR, i will reslove the conflicts and then take a look with the fix.

Reply:
BTW, great catch on CLUSTER SLOTS!

Reply:
@zuiderkwast how did we reproduce it? I can't seem to reproduce it in TCL test.
> If we do CLUSTER SLOTS from EVAL over TLS, it fails at debugServerAssertWithInfo(c, NULL, verifyCachedClusterSlotsResponse(cached_reply, c->resp) == 1); in clusterCommandSlots.

so a simple fix i guess is we use `shouldReturnTlsInfo` to replace the `connIsTLS` in clusterCommandSlots, right? this should be enough i think

Reply:
Yes that's the fix.

I have a test case. I can post it after this pr.

Reply:
great, i merged this one.

Reply:
It's easy to reproduce but we need to build with TLS and run tests with --tls. By default we don't.

Reply:
I did build TLS and runtest with --tls, i suppose i just don't know how to cache a non-tls port repsonse first.